myriad-scheduler/src/main/java/org/apache/myriad/scheduler/fgs/YarnNodeCapacityManager.java

```diff
@@ -135,9 +136,9 @@ private void removeYarnTask(RMContainer rmContainer) {
        */
       state.makeTaskKillable(taskId);
       
-      Node node = retrieveNode(rmContainer);
-      if (node != null) {
-        RMNode rmNode = node.getNode().getRMNode();
+      Optional<Node> optNode = retrieveNode(rmContainer);
+      if (optNode.isPresent()) {
+        RMNode rmNode = optNode.get().getNode().getRMNode();
```

Comment:
While I'm a fan of optionals this usage is a bit overkill, in fact I think it makes this less clear at the cost of extract object creation.